app/models/service_template.rb

```diff
@@ -238,6 +238,11 @@ def template_valid_error_message
   end
 
   def validate_template
+    not_exist = service_resources.detect { |sr| sr.resource.nil? }
+    return {:valid   => false,
+            :message => "Unable to find #{not_exist.resource_type} with Id [#{not_exist.resource_id}]"
+    } if not_exist
```

Comment:
I would prefer we just list all the missing resources instead of just the first one.

Maybe something like this:
```ruby
missing_resources = service_resources.select { |sr| sr.resource.nil? }

if missing_resources.present?
  missing_resource_list = missing_resources.collect {|sr| "#{sr.resource_type}:#{sr.resource_id}"}.join(", ")
  return {:valid   => false,
          :message => "Missing Service Resource(s): #{missing_resource_list}"}
end
```